source/common/http/utility.cc

```diff
@@ -303,6 +303,10 @@ Utility::getLastAddressFromXFF(const Http::HeaderMap& request_headers, uint32_t
     xff_string = xff_string.substr(last_comma + seperator.size());
   }
 
+  // Ignore the whitespace, since they are allowed in HTTP lists (see RFC7239#section-7.1).
+  xff_string = StringUtil::ltrim(xff_string);
+  xff_string = StringUtil::rtrim(xff_string);
```

Comment:
Sorry, just noticed. Do we have a test that covers the need for rtrim?

Reply:
@mattklein123 no... only ltrim(). Lemme modify the case I added so that it has space at both sides...

Reply:
@mattklein123 is fine if I use this test string which should exercise ltrim(), rtrim() and no trim():

```
"192.0.2.10, 192.0.2.1 ,10.0.0.1"
```

Or you rather have me break it up in subtests, and be explicit about what path i am exercising?

Reply:
i guess the above string with comments near each `EXPECT` call should be clear enough... Let me push what I have and lmk what you think.

Reply:
@mattklein123 hold, my comments are in the wrong place, fixing...

Reply:
@mattklein123 should be good now, sorry about that!